txbits/app/models/EngineModel.scala

```diff
@@ -129,19 +129,21 @@ class EngineModel(val db: String = "default") {
   }
 
   def askBid(uid: Option[Long], apiKey: Option[String], base: String, counter: String, amount: BigDecimal, price: BigDecimal, isBid: Boolean) = DB.withConnection(db) { implicit c =>
-    val res = SQL""" select order_new($uid, $apiKey, $base, $counter, ${amount.bigDecimal}, ${price.bigDecimal}, $isBid)
-    """().map(_[Option[Boolean]]("order_new")).head
+    val res = SQL""" select * from order_new($uid, $apiKey, $base, $counter, ${amount.bigDecimal}, ${price.bigDecimal}, $isBid) """().map(row =>
+      row[Option[Long]]("new_id") -> row[Option[BigDecimal]]("new_remains")
+    ).head match {
+      case (Some(id: Long), Some(remains: BigDecimal)) => Some(Json.obj("order" -> id.toString, "remains" -> remains.bigDecimal.toPlainString))
```

Comment:
This is why we have objects that can be serialized to json. Look at EngineModel.scala:306-311 Trade, Withdrawal, etc.